pkg/service/types.go

```diff
@@ -62,8 +68,17 @@ type K8sServiceAccount struct {
 	Name      string
 }
 
-func (ns K8sServiceAccount) String() string {
-	return strings.Join([]string{ns.Namespace, namespaceNameSeparator, ns.Name}, "")
+func (sa K8sServiceAccount) String() string {
+	return strings.Join([]string{sa.Namespace, namespaceNameSeparator, sa.Name}, "")
+}
+
+// GetSyntheticService creates a MeshService for the given K8s Service Account,
+// which has a unique name and is in lieu of an existing Kubernetes service.
+func (sa K8sServiceAccount) GetSyntheticService() MeshService {
+	return MeshService{
+		Namespace: sa.Namespace,
+		Name:      fmt.Sprintf("%s.%s.osm.synthetic-%s", sa.Name, sa.Namespace, SyntheticServiceSuffix),
```

Comment:
Why is the synthetic service associated with a service account? In Kubernetes service is just an abstraction for pods backing it, so would it make more sense to associate a synthetic service to a pod rather than a service account? I guess this is done this way for convenience wrt using custom objects such as `K8sServiceAccount`?

Reply:
SMI Spec uses Service Account for TrafficTarget.
So from there - all you need to do is create a pod (something like [my favorite Jumpod](https://gist.github.com/draychev/16f2ee13206198e3623bc6269789674b#file-jumpod-sh)), add it to an existing Service Account (like Bookstore's for instance), and with this it all works.

I agree - it would be nice to have `pod.GetSyntheticServce()` - but we can't mutate the `pod` struct - so the next best I think is our `K8sServiceAccount`.

We could move this to a standalone function and pass the service account to it.  The goal would be to have one of these services per service account.

Reply:
This is fine, just wanted to understand your thought process.